### ISSUE TITLE
Data type error when reading Win32_VideoController.AdapterRAM

### ISSUE THREAD

**Issue by vnagarnaik** [ohadravid/wmi-rs#35]
When reading `Win32_VideoController`, I am seeing a negative value for `AdapterRAM`. The [Win32_VideoController](https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-videocontroller) doc shows `AdapterRAM` as `uint32`. However, the `wmi` crate sees the data type as `I8`. With 8 bytes, this storage size should be fine for a 4GB card. But the value comes out to a negative number.

I logged the value that is received using a simple query:
```rust
    let wmi_con = WMIConnection::new(COMLibrary::new().unwrap().into()).unwrap();
    let class_obj: Vec<HashMap<String, wmi::Variant>> = wmi_con.raw_query(
        "SELECT Name, AdapterRAM, DriverVersion FROM Win32_VideoController",
    ).unwrap();
    for item in class_obj.iter() {
        info!("Found gpu object:");
        for (key, value) in item.iter() {
            if key == "AdapterRAM" {
                info!("\"{}\": \"{:?}\"", key, value);
            }
        }
    }
```

Output:
```
"AdapterRAM": "I8(-1048576)"
```

This is the same query, but from powershell:
```
PS C:\> Get-WmiObject Win32_VideoController | Select AdapterRAM

AdapterRAM
----------
4293918720
```

I also found a [discussion topic](https://social.technet.microsoft.com/Forums/scriptcenter/en-US/79737c24-547d-4de9-b8da-1c2afc437496/wmi-class-error-with-win32videocontroller-adapter-ram?forum=w8itprogeneral) on this exact field, however, there is no resolution.

Is there anything that can be done to fix this issue within the crate?

**Comment by ohadravid** [comment]
Hi @vnagarnaik ,
It seems like the returned `Variant` really reports itself as `I8` (or `i64`, in Rust).
You can either convert it directly using something like
```
if let Variant::I8(ival) = *value {
    // for ival = -1048576, this prints the expected value.
    println!("i64: {}, u32: {}", ival, ival as u32)
}
```

Or let serde do that for you:
```
#[derive(Deserialize, Debug)]
struct Win32_VideoController {
    Caption: String,
    Name: String,
    AdapterRAM: u32,
}

let results: Vec<Win32_VideoController> = wmi_con.query().unwrap();

println!("{:?}", results);
```
(I can't test this locally since I don't have a >4GB graphic adapter).

I don't know how `Get-WmiObject` can know that, but clearly something fishy is going on if this also happens in VBScript 🤔

**Comment by vnagarnaik** [comment]
Yeah, it's confusing for me as well. I see the negative value as well. The doc reports the field as `u32` but the WMI VARIANT claims it as `i64`. Also there is nothing online which helps with the discrepancy between this raw access and `Get-WmiObject`.

Is there any raw query syntax where we can try to debug what's happening for that field? I have the right hardware, so I can try to gather any logs to help that process.

**Comment by ohadravid** [comment]
The `i64` vs `i32` problem is caused by missing impls [here](https://github.com/ohadravid/wmi-rs/blob/main/src/de/variant_de.rs#L93) - there should be `visit_i{32,16,8},visit_u{32,16,8}` ones as well. You can open a PR to add them and then you'll get the (relatively more?) expected `I4`.

You can try to debug the actual returned variant [here](https://github.com/ohadravid/wmi-rs/blob/main/src/variant.rs#L46), on my machine `variant_type == 3 == VT_I4`, but maybe there are more fields which needs to be inspected?

**Comment by ohadravid** [comment]
Fixed by #36
